File: Data Structures & Algorithms/balanced-binary-tree/submission-10.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public boolean isBalanced(TreeNode root) {
        int [] result = dfs(root);
        return result[0] == 1 ? true : false;
    }


    public int [] dfs(TreeNode root){
        if(root == null){
            return new int []{1, 0};
        }

        int [] left = dfs(root.left);
        int [] right = dfs(root.right);

        boolean isValid = (left[0] == 1 && right[0] == 1 && Math.abs(left[1] - right[1]) <= 1);

        int depth = Math.max(left[1], right[1]) + 1;

        return new int []{isValid ? 1 : 0, depth};
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-10.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:

        left = [0] * len(nums)
        right = [0] * len(nums)
        result = [0] * len(nums)

        temp_value = 1
        for index in range(len(nums)):
            left[index] = temp_value
            temp_value *= nums[index]

        
        temp_value = 1
        for index in reversed(range(len(nums))):
            right[index] = temp_value
            temp_value *= nums[index]
        

        for index in range(len(nums)):
            result[index] = right[index] * left[index]
        
        return result



        